echopype/echodata/combine.py

```diff
@@ -14,170 +14,253 @@
 logger = _init_logger(__name__)
 
 
-def union_attrs(datasets: List[xr.Dataset]) -> Dict[str, Any]:
+def check_echodatas_input(echodatas: List[EchoData]) -> Tuple[str, List[str]]:
     """
-    Merges attrs from a list of datasets.
-    Prioritizes keys from later datasets.
+    Ensures that the input ``echodatas`` for ``combine_echodata``
+    is in the correct form and all necessary items exist.
+
+    Parameters
+    ----------
+    echodatas: List[EchoData]
+        The list of `EchoData` objects to be combined.
+
+    Returns
+    -------
+    sonar_model : str
+        The sonar model used for all values in ``echodatas``
+    echodata_filenames : List[str]
+        The source files names for all values in ``echodatas``
     """
 
-    total_attrs = dict()
-    for ds in datasets:
-        total_attrs.update(ds.attrs)
-    return total_attrs
+    sonar_model = None
+    echodata_filenames = []
+    for ed in echodatas:
 
+        # check sonar model and store it
+        if ed.sonar_model is None:
+            raise ValueError("all EchoData objects must have non-None sonar_model values")
+        elif sonar_model is None:
+            sonar_model = ed.sonar_model
+        elif ed.sonar_model != sonar_model:
+            raise ValueError("all EchoData objects must have the same sonar_model value")
```

Comment:
I think `sonar_model` should be assigned to be the first `ed.sonar_model` outside of the loop, and the if-else can be simplified:
```
        if ed.sonar_model is None:
            raise ValueError("all EchoData objects must have non-None sonar_model values")
        elif ed.sonar_model != sonar_model:
            raise ValueError("all EchoData objects must have the same sonar_model value")
```
It does not make sense to check if `sonar_model` is None within the loop.
I think these code snippets are from before and we can correct this now.

Reply:
Yes, this is a code snippet from before. I agree, it is not necessary. I will check and set `sonar_model` at the beginning of the loop.